src/components/PostsList.tsx

```diff
@@ -1,56 +1,53 @@
 import classNames from 'classnames';
 import React from 'react';
-import { Post } from '../types/Post';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { set } from '../features/selectedPostSlice';
 
-type Props = {
-  posts: Post[],
-  selectedPostId?: number,
-  onPostSelected: (post: Post | null) => void,
-};
-
-export const PostsList: React.FC<Props> = ({
-  posts,
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+export const PostsList: React.FC = () => {
+  const posts = useAppSelector(state => state.posts.items);
+  const selectedPost = useAppSelector(state => state.selectedPost.selectedPost);
```

Comment:
1. I suggest to create and export all selectors callbacks from certain slice....ts file
2. It looks a little strange `selectedPost.selectedPost`, try to avid embeddedness like this

```suggestion
// postsSlice.ts
... 
 export const selectPosts = state => state.posts.items;
 export const selectSelectedPost = state => state.selectedPost.selectedPost;
 
export const { set } = postsSlice.actions;
export default postsSlice.reducer;

 // component.tsx
 ...
  const posts = useAppSelector(selectPosts);
  const selectedPost = useAppSelector(selectSelectedPost );
...
```